File: Apps/Scripts/Pyinstaller/EfazRobloxBootstrapPlayRoblox.py
import sys
import os
import subprocess
import platform

def printMainMessage(mes):
    print(f"\033[38;5;255m{mes}\033[0m")

def printErrorMessage(mes):
    print(f"\033[38;5;196m{mes}\033[0m")

def printSuccessMessage(mes):
    print(f"\033[38;5;82m{mes}\033[0m")

def printWarnMessage(mes):
    print(f"\033[38;5;202m{mes}\033[0m")

def printDebugMessage(mes):
    print(f"\033[38;5;226m{mes}\033[0m")

if __name__ == "__main__":
    current_version = {"version": "1.2.0"}
    main_os = platform.system()
    direct_run = False
    args = sys.argv

    printWarnMessage("-----------")
    printWarnMessage("Welcome to Efaz's Roblox Bootstrap Loader!")
    printWarnMessage("Made by Efaz from efaz.dev!")
    printWarnMessage(f"v{current_version['version']}")
    printWarnMessage("-----------")
    printMainMessage("Determining System OS..")

    if main_os == "Darwin":
        if os.path.exists("/Applications/EfazRobloxBootstrap.app/"):
            url_scheme_path = "/Applications/EfazRobloxBootstrap.app/Contents/Resources/URLSchemeExchange"
            with open(url_scheme_path, "w") as f:
                f.write("efaz-bootstrap://continue")
            printMainMessage(f"Created URL Exchange File: {url_scheme_path}")
            printMainMessage("Loading EfazRobloxBootstrap executable!")
            result = subprocess.run(["open", "-n", "-a", "/Applications/EfazRobloxBootstrap.app/Contents/MacOS/EfazRobloxBootstrap"], cwd="/Applications/EfazRobloxBootstrap.app/Contents/Resources/", stdout=subprocess.PIPE, stderr=subprocess.PIPE, check=True)
            if result.returncode == 0:
                printSuccessMessage(f"Bootstrap Run Success: {result.returncode}")
            else:
                printErrorMessage(f"Bootstrap Run Failed: {result.returncode}")
            sys.exit(0)
        else:
            printErrorMessage("Bootstrap Run Failed: App is not installed.")
    elif main_os == "Windows":
        generated_app_path = os.path.join(os.getenv('LOCALAPPDATA'), "EfazRobloxBootstrap")
        if os.path.exists(os.path.join(generated_app_path, "EfazRobloxBootstrap.exe")):
            url_scheme_path = os.path.join(generated_app_path, "URLSchemeExchange")
            with open(url_scheme_path, "w") as f:
                f.write("efaz-bootstrap://continue")
            printMainMessage(f"Created URL Exchange File: {url_scheme_path}")
            printMainMessage("Loading EfazRobloxBootstrap.exe!")
            os.system(f'start {os.path.join(generated_app_path, "EfazRobloxBootstrap.exe")}')
            sys.exit(0)
        else:
            printErrorMessage("Bootstrap Run Failed: App is not installed.")
else:
    class EfazRobloxBootstrapNotModule(Exception):
        def __init__(self):            
            super().__init__("Efaz's Roblox Bootstrap is only a runable instance, not a module.")
    class EfazRobloxBootstrapInstallerNotModule(Exception):
        def __init__(self):            
            super().__init__("The installer for Efaz's Roblox Bootstrap is only a runable instance, not a module.")
    class EfazRobloxBootstrapLoaderNotModule(Exception):
        def __init__(self):            
            super().__init__("The loader for Efaz's Roblox Bootstrap is only a runable instance, not a module.")
    raise EfazRobloxBootstrapLoaderNotModule()

File: Apps/Scripts/Clang/EfazRobloxBootstrapPlayRoblox.cpp
#include <iostream>
#include <fstream>
#include <string>
#include <cstdlib>
#include <cstdio>
#include <filesystem>
#include <vector>

#ifdef _WIN32
#include <windows.h>
#endif

void printMainMessage(const std::string& mes) {
    std::cout << "\033[38;5;255m" << mes << "\033[0m" << std::endl;
}

void printErrorMessage(const std::string& mes) {
    std::cout << "\033[38;5;196m" << mes << "\033[0m" << std::endl;
}

void printSuccessMessage(const std::string& mes) {
    std::cout << "\033[38;5;82m" << mes << "\033[0m" << std::endl;
}

void printWarnMessage(const std::string& mes) {
    std::cout << "\033[38;5;202m" << mes << "\033[0m" << std::endl;
}

int main(int argc, char* argv[]) {
    std::string current_version = "1.2.3";
    std::string main_os;
    
    #ifdef __APPLE__
        main_os = "Darwin";
    #elif _WIN32
        main_os = "Windows";
    #else
        main_os = "Other";
    #endif

    printWarnMessage("-----------");
    printWarnMessage("Welcome to Efaz's Roblox Bootstrap Loader!");
    printWarnMessage("Made by Efaz from efaz.dev!");
    printWarnMessage("v" + current_version);
    printWarnMessage("-----------");
    printMainMessage("Determining System OS...");

    if (main_os == "Darwin") {
        std::string app_path = "/Applications/EfazRobloxBootstrap.app/";
        if (std::filesystem::exists(app_path)) {
            std::string url_scheme_path = app_path + "Contents/Resources/URLSchemeExchange";
            std::ofstream file(url_scheme_path);
            if (file.is_open()) {
                file << "efaz-bootstrap://continue";
                file.close();
            }
            printMainMessage("Created URL Exchange File: " + url_scheme_path);
            printMainMessage("Loading EfazRobloxBootstrap executable!");
            int result = std::system("open -n -a /Applications/EfazRobloxBootstrap.app/Contents/MacOS/EfazRobloxBootstrap");
            if (result == 0) {
                printSuccessMessage("Bootstrap Run Success: " + std::to_string(result));
            } else {
                printErrorMessage("Bootstrap Run Failed: " + std::to_string(result));
            }
            return 0;
        } else {
            printErrorMessage("Bootstrap Run Failed: App is not installed.");
        }
    } else if (main_os == "Windows") {
        char* localAppData = std::getenv("LOCALAPPDATA");
        if (localAppData) {
            std::string generated_app_path = std::string(localAppData) + "\\EfazRobloxBootstrap";
            std::string exe_path = generated_app_path + "\\EfazRobloxBootstrap.exe";
            if (std::filesystem::exists(exe_path)) {
                std::string url_scheme_path = generated_app_path + "\\URLSchemeExchange";
                std::ofstream file(url_scheme_path);
                if (file.is_open()) {
                    file << "efaz-bootstrap://continue";
                    file.close();
                }
                printMainMessage("Created URL Exchange File: " + url_scheme_path);
                printMainMessage("Loading EfazRobloxBootstrap.exe!");
                std::string command = "start " + exe_path;
                std::system(command.c_str());
                return 0;
            } else {
                printErrorMessage("Bootstrap Run Failed: App is not installed.");
            }
        }
    } else {
        printMainMessage("Efaz's Roblox Bootstrap is not a module.");
        std::cin.get();
        return 1;
    }
    return 0;
}
